sysboundary/ionosphere.cpp

```diff
@@ -535,61 +535,49 @@ namespace SBC {
     */
    Real Ionosphere::fieldSolverBoundaryCondMagneticField(
       FsGrid< std::array<Real, fsgrids::bfield::N_BFIELD>, 2> & perBGrid,
-      FsGrid< std::array<Real, fsgrids::bfield::N_BFIELD>, 2> & perBDt2Grid,
-      FsGrid< std::array<Real, fsgrids::efield::N_EFIELD>, 2> & EGrid,
-      FsGrid< std::array<Real, fsgrids::efield::N_EFIELD>, 2> & EDt2Grid,
       FsGrid< fsgrids::technical, 2> & technicalGrid,
       cint i,
       cint j,
       cint k,
       creal& dt,
-      cuint& RKCase,
       cuint& component
    ) {
-      FsGrid< std::array<Real, fsgrids::bfield::N_BFIELD>, 2> * bGrid;
-      
-      if(RKCase == RK_ORDER1 || RKCase == RK_ORDER2_STEP2) {
-         bGrid = &perBGrid;
-      } else {
-         bGrid = &perBDt2Grid;
-      }
-      
       if (technicalGrid.get(i,j,k)->sysBoundaryLayer == 1) {
          switch(component) {
             case 0:
                if (  ((technicalGrid.get(i-1,j,k)->SOLVE & compute::BX) == compute::BX)
                   && ((technicalGrid.get(i+1,j,k)->SOLVE & compute::BX) == compute::BX)
                ) {
-                  return 0.5 * (bGrid->get(i-1,j,k)->at(fsgrids::bfield::PERBX) + bGrid->get(i+1,j,k)->at(fsgrids::bfield::PERBX));
+                  return 0.5 * (perBGrid.get(i-1,j,k)->at(fsgrids::bfield::PERBX) + perBGrid.get(i+1,j,k)->at(fsgrids::bfield::PERBX));
```

Comment:
Here and on lines below it'd actually be better to keep it as bGrid, as it can in fact be perBGrid or perBDt2Grid, right?

And make the grid received in the function call be bGrid, of course.

Reply:
That'd make this diff easier to follow indeed. Thanks.